Singularity

```diff
@@ -3,6 +3,8 @@ From: continuumio/anaconda3
 
 %runscript
 
+     echo "Mounting shared drive..."
+     /sbin/mount.cifs //server/share /opt/notebooks/share -o user=shareusername,password=sharepassword
```

Comment:
Could this be passed in to the container from the host running it?

E.g.,

```bash
shareuser=${1}
sharepass=${2}

...
/sbin/mount.cifs //server/share /opt/notebooks/share -o user=${shareuser},password=${sharepass}
```

of course you would want to do a better job of checking that both are defined and then spitting out an error if not. Generally, I would never suggest or store any kind of credential in a git repo, so I wouldn't want to suggest for the user to do it. Thoughts?

Reply:
Good idea, I'll have a go at it.

Reply:
Got stuck quickly -- how does one pass parameters to the runscript when using a sandbox and not an executable image?
`sudo singularity run --writable jupyter-box server share user pw` doesn't work

Reply:
It should be exactly the same! If you want to post what you have so far I can help tomorrow, going to bed soon :)

Reply:
regardless of what arguments I add to that line, I get: 
```bash
/.singularity.d/runscript: 4: /.singularity.d/runscript: server: not found
/.singularity.d/runscript: 5: /.singularity.d/runscript: share: not found
/.singularity.d/runscript: 6: /.singularity.d/runscript: shareuser: not found
/.singularity.d/runscript: 7: /.singularity.d/runscript: sharepw: not found
Mounting shared drive /// with username
mount.cifs: bad UNC (///)
Starting notebook...
```

Reply:
```diff
diff --git a/Singularity b/Singularity
index 694fa50..36596d5 100644
--- a/Singularity
+++ b/Singularity
@@ -3,8 +3,12 @@ From: continuumio/anaconda3

 %runscript

-     echo "Mounting shared drive..."
-     /sbin/mount.cifs //server/share /opt/notebooks/share -o user=shareusername,password=sharepassword
+     server = ${1}
+     share = ${2}
+     shareuser = ${3}
+     sharepw = ${4}
+     echo "Mounting shared drive //${server}/${share} with username ${shareuser}"
+     /sbin/mount.cifs //${server}/${share} /opt/notebooks/share -o user=${shareuser},password=${sharepw}
      echo "Starting notebook..."
      echo "Open browser to localhost:8888"
      exec /opt/conda/bin/jupyter notebook --notebook-dir=/opt/notebooks --ip='*' --port=8888 --no-browser --allow-root
```

Reply:
You can’t have spaces with bash variables, that’s the issue.

Reply:
Also it’s best practice to put them in double quotes.